### PR TITLE
feat: Strapi API Integration

### PR DESCRIPTION
**Pull request by naseemshah**
# Explain the feature/fix

This PR includes API integration from Strapi backend and partial user authentication is implemented (Integrated Google Login Button and Receiving JWT token from Strapi)

API Integrated:
- KSTRC
- Destination Page
- Activity Page
- Review Page (only retrieving reviews, Adding Review feature after complete implementation of authentication)

### CODE REVIEW

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: src/components/Common/Heading.js [hunk 1]
```diff
@@ -3,30 +3,44 @@ import Star from "./Star";
...
+        moderate: "text-yellow-500 border-yellow-500",
+        difficult: "text-red-500 border-red-500",
+    };
+
+    let getAvailableDays = (days) => {
+        let availableDays = Object.keys(days).filter(
+            (key) => days[key] === true
+        );
+
+        let formatedAvailableDays = [];
```
Comment (lespaceman) [inline src/components/Common/Heading.js:34]: let formatedAvailableDays = availableDays.map(...

File: src/components/Common/Rails/Carousal/CarousalCard.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                JSON.stringify(currentDestinationFavourites)
+            );
+        } else if (currentDestinationFavourites && !liked) {
+            currentDestinationFavourites.push(id);
+            localStorage.setItem(
+                "hello-munnar-destination-favourites",
+                JSON.stringify(currentDestinationFavourites)
+            );
+        } else if (!currentDestinationFavourites) {
+            console.log("here");
```
Comment (lespaceman) [inline src/components/Common/Rails/Carousal/CarousalCard.js:50]: Remove console.log()

File: src/components/User/SignIn.js [hunk 3]
```diff
@@ -1,27 +1,66 @@
...
+        });
+    };
+
+    let authBackend = async (accessToken) => {
+        return api
+            .get(`/auth/google/callback?access_token=${accessToken}`)
+            .then((response) => {
+                let { status, data } = response;
+                if (status === 200) {
+                    console.log(data);
```
Comment (lespaceman) [inline src/components/User/SignIn.js:31]: remove console.log()

File: src/router/PublicRouter.js [hunk 4]
```diff
@@ -39,13 +40,20 @@ const routes = {
...
+    // let { authInit } = useContext(UserContext);
+
     const pagesToShowNavBar = ["/", "/favourites", "/nearme", "/experiences"];
     const path = usePath();
-    React.useEffect(() => {
-        // Send page view event
-        ReactGA.set({ page: path });
-        ReactGA.pageview(window.location.pathname + window.location.search);
-    }, [path]);
+    // useEffect(() => {
```
Comment (lespaceman) [inline src/router/PublicRouter.js:47]: remove commented code

File: src/components/Ksrtc/KsrtcContainer.js [hunk 5]
```diff
@@ -1,25 +1,69 @@
...
+    }, []);
+
+    let getKSRTCs = useCallback(async () => {
+        setIsLoading(true);
+        await api
+            .get(`/ksrtcs`)
+            .then((response) => {
+                let { status, data } = response;
+                if (status === 200) {
+                    console.log(data);
```
Comment (lespaceman) [inline src/components/Ksrtc/KsrtcContainer.js:23]: remove console.log()

**Review by gautamkrishnar — CHANGES_REQUESTED** [review]

**Review by lespaceman — APPROVED** [review]

**Review by gautamkrishnar — APPROVED** [review]

**Pull request merged**